pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -237,12 +239,15 @@ async def run(
             for tool in self._function_tools.values():
                 tool.current_retry = 0
 
-            usage = result.Usage()
-
+            usage = result.Usage(requests=0)
             model_settings = merge_model_settings(self.model_settings, model_settings)
+            usage_limits = usage_limits or UsageLimits()
 
             run_step = 0
             while True:
+                usage.requests += 1
```

Comment:
I think this is the wrong place to increment `requests`, should be after a request, same place as `usage += request_usage`

Reply:
Okay, I’m assuming we still want to error before making the request though, right? Rather than after the limit is exceeded. (I can do that without incrementing here, just want to confirm.)

Reply:
well, if the request limit is 2, we should error some time after request number 2 and before request number 3.